### PR TITLE
[React] Hooks (logout component)

### PR DESCRIPTION
**Pull request by qmonmert**
WIP on #9891

-   [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by deepu105** [comment]
Could you check if the failures are actual errors? also aren't you able to
to restart onloy the failed build from Travis?

Thanks & Regards,
Deepu

**Comment by qmonmert** [comment]
@deepu105 I don't understand the errors, the error is on Azure

**Comment by deepu105** [comment]
@qmonmert any update on this one. Do you need any help?

**Comment by qmonmert** [comment]
@deepu105 I need help because on my machine all tests pass :/

**Comment by pascalgrimaud** [comment]
@qmonmert : did you test locally with prod profile ?

**Comment by qmonmert** [comment]
@pascalgrimaud I just tried and it works locally :/

File: generators/client/templates/react/src/main/webapp/app/modules/login/logout.tsx.ejs [hunk 1]
```diff
@@ -27,29 +27,26 @@ export interface ILogoutProps extends StateProps, DispatchProps {
...
-  componentDidMount() {
-    this.props.logout();
+  const logoutUrl = props.logoutUrl;
+  if (logoutUrl) {
+    // if Keycloak, logoutUrl has protocol/openid-connect in it
+    window.location.href =
+      logoutUrl.includes('/protocol')
+        ? logoutUrl + '?redirect_uri=' + window.location.origin
+        : logoutUrl + '?id_token_hint=' + props.idToken + '&post_logout_redirect_uri=' + window.location.origin;
   }
```
Comment (SoniaLahi) [inline generators/client/templates/react/src/main/webapp/app/modules/login/logout.tsx.ejs]: @qmonmert: wouldn't it be better to add these lines of code in the useEffect function, as the actions on a component lifecycle event are specified in this function ?

**Comment by pascalgrimaud** [comment]
@qmonmert : first of all, sorry for taking time to review this.

I just gave a try locally and it failed, exactly with same error than in CI.
So the error comes from this PR, and not from Azure.
Then, it failed during entity e2e tests, so it's really strange...

I pushed a repo with the same config than Azure, so you can use it:

```
git clone https://github.com/pascalgrimaud/jh-react-pr-9968
cd jh-react-pr-9968
npm install
./mvnw clean verify -Pprod
docker-compose -f src/main/docker/mysql.yml up -d
java -jar target/*.jar --spring.profiles.active=prod,swagger
```

Then, run: `npm run e2e` in another terminal.

It looks like after a delay, the logout doesn't work anymore, so we have something like this:

```
FieldTestInfiniteScrollEntity e2e test
    1) "before all" hook: ret for "should load FieldTestInfiniteScrollEntities"
    2) "after all" hook: ret for "should delete last FieldTestInfiniteScrollEntity"

  FieldTestMapstructEntity e2e test
    3) "before all" hook: ret for "should load FieldTestMapstructEntities"
    4) "after all" hook: ret for "should delete last FieldTestMapstructEntity"

  FieldTestPagerEntity e2e test
    5) "before all" hook: ret for "should load FieldTestPagerEntities"
    6) "after all" hook: ret for "should delete last FieldTestPagerEntity"

  FieldTestPaginationEntity e2e test
    7) "before all" hook: ret for "should load FieldTestPaginationEntities"
    8) "after all" hook: ret for "should delete last FieldTestPaginationEntity"

```

**Comment by pascalgrimaud** [comment]
@qmonmert : did you manage to reproduce the issue locally ?

Then, I think we need some React experts to help for achieving this PR.

@sendilkumarn @deepu105 ?

**Comment by deepu105** [comment]
@pascalgrimaud @qmonmert I need to find some time to loom at it, also its been a while since I did any serious react, next two weeks i'll be busy with Devoxx Ukraine and Codemotion Berlin but after that I can see whats going on here

**Comment by deepu105** [comment]
These kinds of issues are mostly due to some race conditions in our code which become apparent when protractor runs through things in superhuman speeds, it would be less evident when we use those pages as we are normally slow in those interactions. @qmonmert maybe try using some wait in between the actions in the protractor tests to see if this really is due to race conditions

**Comment by pascalgrimaud** [comment]
@qmonmert : it looks ok, finally !!!

**Comment by qmonmert** [comment]
@pascalgrimaud I found a solution but continuous-integration is always red 😕

**Comment by pascalgrimaud** [comment]
@qmonmert : the failed build is related to relationships, there is a ticket for that. Nothing related to your PR. As it looks OK, I'm merging this.
Thanks for your patience, still 1 last PR !!

**Comment by qmonmert** [comment]
@pascalgrimaud 🙂 I did not see the last failed 😇 so I have had a genius idea this night !

**Pull request merged**